lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -167,10 +167,11 @@ def add_extra_data(post, payment, options)
       end
 
       def add_risk_data(post, options)
-        risk_data = {}
-        risk_data.merge!(options[:risk_data]) if options[:risk_data]
+        risk_data = options.fetch(:risk_data, {})
+        return nil if risk_data.empty?
+        risk_data = Hash[risk_data.map { |k, v| ["riskdata.#{k}", v] }]
 
-        post[:additionalData][:riskData] = risk_data unless risk_data.empty?
+        post[:additionalData].merge!(risk_data)
```

Comment:
It occurs to me that this method can be further simplified since we're not creating a sub-hash under `additionalData`. It makes sense to exit early and avoid creating an empty sub-hash, but if we're merging directly into `additionalData` I'm pretty sure it's neither more efficient than simply merging the hash (even if it's empty). I'm going to tweak this a bit and see if I can't simplify. Won't take much effort to do.

Reply:
Would the following be preferable to avoid needless instantiation of a new `Hash` when not needed?

```
      def add_risk_data(post, options)
        if (risk_data = options[:risk_data])
          risk_data = Hash[risk_data.map { |k, v| ["riskdata.#{k}", v] }]
          post[:additionalData].merge!(risk_data)
        end
      end
```

Reply:
@curiousepic When you have a moment, would you give me your opinion on this? I don't want to try to micro-optimize here, but when I thought about it more it seemed more logical to just skip all of it at least if the user doesn't pass any `risk_data` option at all. Since you already approved, I don't want to sneak in a change after-the-fact without running it by you first. Thanks!

Reply:
Sure, that's reasonable.